flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/StringToBinaryCastRule.java

```diff
@@ -45,6 +52,25 @@ public String generateExpression(
             String inputTerm,
             LogicalType inputLogicalType,
             LogicalType targetLogicalType) {
-        return methodCall(inputTerm, "toBytes");
+        int inputLength = Integer.MAX_VALUE;
```

Comment:
You can use `int inputLength = LogicalTypeChecks.getLength(inputLogicalType)` directly without `instanceof`.

Reply:
Got it, thanks!